ibis/backends/duckdb/__init__.py

```diff
@@ -26,6 +27,50 @@ class _ColumnMetadata(NamedTuple):
     type: dt.DataType
 
 
+_register = RegexDispatcher("register")
```

Comment:
While this RegexDispatcher is very cool, why not just use `urlparse` as get the scheme?

Reply:
Two reasons:

1. We need to handle file paths as well, and the code to do that is more readable if we don't mix URI handling and path handling in the same block of code.
2. We're going to reuse this in another API `ibis.connect` that also needs to handle both URIs and paths, and we'd end up either duplicating the logic or extracting it out into a similar amount of code.